esmvaltool/diag_scripts/portrait_plot.py

```diff
@@ -0,0 +1,554 @@
+            linewidth=0.5,
+            fill=True,
+        )
+        axe.add_patch(patch)
+
+
+def plot_group(cfg, axe, data, title=None):
+    """Create matrix for one subplot in ax using plt.imshow.
+
+    by default split None is used, if all splits are named the first is
+    used. Other splits will be added by overlaying triangles.
+    """
+    split = data.sel({cfg["split_by"]: cfg["default_split"]})
+    plot_matrix(
+        split.values.T,  # 2d numpy array
+        split.coords[cfg["y_by"]].values,  # y_labels
+        split.coords[cfg["x_by"]].values,  # x_labels
+        axe,
+        cfg["plot_kwargs"],
+    )
+    if title is not None:
+        axe.set_title(title)
+    axe.set(**cfg["axes_properties"])
+
+
+def get_triangle_nodes(position, total_count=2):
+    """Return list of nodes with relative x, y coordinates.
+
+    The nodes of the triangle are given as list of three tuples. Each tuples
+    contains relative coordinates (-0.5 to +0.5). For total of <= 2 a top left
+    (position=0) and bottom right (position=1) rectangle is returned.
+    For higher counts (3 or 4) one quartile is returned for each position.
+    NOTE: Order matters. Ensure axis labels for the legend match when changing.
+    """
+    if total_count < 3:
+        halves = [
+            [(0.5, -0.5), (-0.5, -0.5), (-0.5, 0.5)],  # top left
+            [(0.5, -0.5), (0.5, 0.5), (-0.5, 0.5)],  # bottom right
+        ]
+        return halves[position]
+    quarters = [
+        [(-0.5, -0.5), (0, 0), (-0.5, 0.5)],  # left
+        [(0.5, -0.5), (0, 0), (0.5, 0.5)],  # right
+        [(-0.5, 0.5), (0, 0), (0.5, 0.5)],  # bottom
+        [(-0.5, -0.5), (0, 0), (0.5, -0.5)],  # top
+    ]
+    return quarters[position]
+
+
+def plot_overlays(cfg, grid, data):
+    """Call overlay_reference for each split in data and each group in grid."""
+    split_count = data.shape[3]
+    group_count = data.shape[2]
+    for i in range(group_count):
+        if split_count < 2:
+            log.debug("No additional splits for overlay.")
+            break
+        if split_count > 4:
+            log.warning("Too many splits for overlay, only 3 will be plotted.")
+        group_data = data.isel({cfg["group_by"]: i})
+        group_data = group_data.dropna(cfg["x_by"], how="all")
+        for sss in range(split_count):
+            split = group_data.isel({cfg["split_by"]: sss})
+            split_label = split.coords[cfg["split_by"]].values.item()
+            if split_label == cfg["default_split"]:
+                log.debug("Skipping default split for overlay.")
+                continue
+            nodes = get_triangle_nodes(sss, split_count)
+            overlay_reference(cfg, grid[i], split.values.T, nodes)
+
+
+def plot(cfg, data):
+    """Create figure with subplots for each group and save to NetCDF.
+
+    sets same color range and  overlays additional references based on
+    the content of data (xr.DataArray)
+    """
+    save_to_netcdf(cfg, data)
```

Comment:
```suggestion

```
Move this outside the plotscript into main to disentable plotting and saving the data.